### PR TITLE
Highlight read

### PR DESCRIPTION
**Pull request by Shin-703**
This is an answer to the issue #7 highlight read. This enhancement adds highlights when explicitly reading data (in data segment or register block) while executing step by step.

There is now two colors for these highlights (configurable by the user), blue for explicit reading and redish for explicit writing.

### CODE REVIEW

**Comment by privat** [comment]
There are some errors on my machine, maybe some settings are not initialized properly

```
$ java -jar rars.jar 
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException: Cannot invoke "java.awt.Color.equals(Object)" because "<local3>" is null
	at rars.venus.settings.SettingsHighlightingAction.initializeButtonColors(SettingsHighlightingAction.java:341)
	at rars.venus.settings.SettingsHighlightingAction.buildDialogPanel(SettingsHighlightingAction.java:175)
```

**Comment by privat** [comment]
it works \o/ :tada:

File: src/rars/venus/registers/RegisterBlockWindow.java [hunk 1]
```diff
@@ -162,14 +164,42 @@ public void updateRegisters() {
      * @param register Register object corresponding to row to be selected.
      */
     private void highlightCellForRegister(Register register) {
-        for (int i = 0; i < registers.length; i++) {
+        /*for (int i = 0; i < registers.length; i++) {
```
Comment (privat) [inline src/rars/venus/registers/RegisterBlockWindow.java:167]: do not comment, delete

File: src/rars/venus/DataSegmentWindow.java [hunk 2]
```diff
@@ -1033,11 +1044,16 @@ public Component getTableCellRendererComponent(JTable table, Object value,
...
+            if (settings.getBooleanSetting(Settings.Bool.EXPLICIT_WRITE_HIGHLIGHTING) && addressHighlighting &&
+                    rowFirstAddress == addressRowFirstAddress && column == addressColumn && writingHighlight) {
+                cell.setBackground(settings.getColorSettingByPosition(Settings.EXPLICIT_WRITE_HIGHLIGHT_BACKGROUND));
+                cell.setForeground(settings.getColorSettingByPosition(Settings.EXPLICIT_WRITE_HIGHLIGHT_FOREGROUND));
+                cell.setFont(settings.getFontByPosition(Settings.EXPLICIT_WRITE_HIGHLIGHT_FONT));
+            } else if (settings.getBooleanSetting(Settings.Bool.EXPLICIT_READ_HIGHLIGHTING) && addressHighlighting &&
+                    rowFirstAddress == addressRowFirstAddress && column == addressColumn && !writingHighlight) { //not writing highlight means reading highlight
+                    cell.setBackground(settings.getColorSettingByPosition(Settings.EXPLICIT_READ_HIGHLIGHT_BACKGROUND));
+                    cell.setForeground(settings.getColorSettingByPosition(Settings.EXPLICIT_READ_HIGHLIGHT_FOREGROUND));
+                    cell.setFont(settings.getFontByPosition(Settings.EXPLICIT_READ_HIGHLIGHT_FONT));
```
Comment (privat) [inline src/rars/venus/DataSegmentWindow.java:1056]: weird indentation

**Review by privat — CHANGES_REQUESTED** [review]

File: src/rars/venus/DataSegmentWindow.java (hunk 2 above)
Comment (Shin-703) [reply src/rars/venus/DataSegmentWindow.java]: yes indeed

**Pull request merged**